marklogic-data-hub-central/ui/e2e/cypress/integration/modeling/graphValidations.spec.tsx

```diff
@@ -161,4 +161,49 @@ describe("Graph Validations", () => {
       graphViewSidePanel.getDeleteIcon("Person").should("be.visible");
     });
   });
+
+  it("can select all available nodes in graph view, locations persist", () => {
+    let ids = ["BabyRegistry", "Client", "Customer", "Order", "Person"];
+    let savedCoords: any = {};
+    modelPage.selectView("project-diagram");
+
+    // Select each entity node using retrieved coords
+    ids.forEach(id => {
+      graphVis.getPositionsOfNodes(id).then((nodePositions: any) => {
+        let coords: any = nodePositions[id];
+        savedCoords[id] = {x: coords.x, y: coords.y};
+        graphVis.getGraphVisCanvas().click(coords.x, coords.y);
+      });
+      // Verify entity is shown in side panel and close
+      graphViewSidePanel.getSelectedEntityHeading(id).should("be.visible");
+      graphViewSidePanel.closeSidePanel();
+    });
+
+    // Exit graph view and return
+    cy.waitUntil(() => toolbar.getCurateToolbarIcon()).click();
+    cy.waitUntil(() => toolbar.getModelToolbarIcon()).click();
+    modelPage.selectView("project-diagram");
+
+    // Select entity nodes using previously saved coords
+    ids.forEach(id => {
+      graphVis.getPositionsOfNodes(id).then(() => {
```

Comment:
Look like we are using the previously stored positions only instead of positions from output of getPositionsOfNodes(). In this case we can avoid firing `graphVis.getPositionsOfNodes(id)` and instead, let's place the click action (`graphVis.getGraphVisCanvas().click(savedCoords[id].x, savedCoords[id].y)`) directly under the forEach loop.

Reply:
@xnikhil08 and I looked into this fix and it's not straightforward since removing that async wrapper in the second forEach loops breaks the async dependency from the first loop (savedCoords is not saved by the time the second loop is executed). We agreed to keep it as-is for now, I will add a TODO note.